coretex/cli/modules/utils.py

```diff
@@ -15,22 +15,53 @@
 #     You should have received a copy of the GNU Affero General Public License
 #     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-from typing import List, Any, Optional, Callable
+from typing import List, Any, Tuple, Optional, Callable
 from functools import wraps
+from importlib.metadata import version as getLibraryVersion
 
 import sys
+import venv
 
 from py3nvml import py3nvml
 
 import click
+import platform
 
+from ...configuration import DEFAULT_VENV_PATH
 from ...utils.process import command
 
 
+def checkEnvironment() -> None:
+    venvPython = DEFAULT_VENV_PATH / "bin" / "python"
+    if DEFAULT_VENV_PATH.exists():
+        return
+
+    venv.create(DEFAULT_VENV_PATH, with_pip = True)
+
+    if platform.system() == "Windows":
+        venvPython = DEFAULT_VENV_PATH / "Scripts" / "python.exe"
+
+    currentCtxVersion = getLibraryVersion("coretex")
+    command([str(venvPython), "-m", "pip", "install", f"coretex=={currentCtxVersion}"], ignoreStdout = True)
```

Comment:
This will install production CLI from PyPi, what if someone installed it using git url?

Reply:
I get the point but should we bother with that now?